lib/bulkrax.rb

```diff
@@ -168,4 +169,43 @@ def self.multi_value_element_split_on
   def self.setup
     yield self
   end
+
+  # Responsible for stripping hidden characters from the given string.
+  #
+  # @param value [#to_s]
+  # @return [String] with hidden characters removed
+  #
+  # @see https://github.com/samvera-labs/bulkrax/issues/688
+  def self.normalize_string(value)
+    # Removing [Byte Order Mark (BOM)](https://en.wikipedia.org/wiki/Byte_order_mark)
+    value.to_s.delete("\xEF\xBB\xBF")
+  end
+
+  # This class confirms to the Active::Support.serialze interface.  It's job is to ensure that we
+  # don't have keys with the tricksy Byte Order Mark character.
+  #
+  # @see https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html#method-i-serialize
+  class NormalizedJson
+    def self.normalize_keys(hash)
+      return hash unless hash.respond_to?(:each_pair)
+      returning_value = {}
+      hash.each_pair do |key, value|
+        returning_value[Bulkrax.normalize_string(key)] = value
+      end
+      returning_value
+    end
+
+    # Serializes an attribute value to a string that will be stored in the database.
+    def self.dump(value)
+      JSON.dump(normalize_keys(value))
+    end
```

Comment:
1. this comment leads me to believe we're passing in a value only, and not the entire property as a key/value paired hash. is that correct? if so, wouldn't we immediately return the value from `normalize_keys` because of line 190?
2. where is this called from? or is this setup for the future?

Reply:
The Rail's [ActiveRecord::Base.serialize](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html#method-i-serialize) takes one or more arguments.

The first argument is the symbol for the attribute we are serializing (e.g. `:raw_metadata`).  If we provide a second parameter that is the `class_name_or_coder`.  In this case we are wanting to use the `Bulkrax::NormalizedJson`.  The `.dump` method is what we use to coerce input before writing that attribute to the database.  The `.load` is what we use when reading from the database.

Previously we had `serialize :raw_metadata, JSON` which loaded and dumped the input by first converting it to JSON.  With this switch, we're using `serialize :raw_metadata, Bulkrax::NormalizedJson` which will continue to use JSON as the serialization format in the database but will also ensure that the keys don't have a Byte Order Mark in them.

Reply:
To the second question, there is not future setup.

Reply:
ahhhhh. knowing that `value` = `:raw_metadata` makes sense now. thanks!